### PR TITLE
[source-mysql] Adding CAT tests for source-mysql

### PR DESCRIPTION
**Pull request by theyueli**
## What

This patch implements CAT tests for source-mysql. 

The implementation will be reused for testing source-mysql-v2. 

This patch contributes to https://github.com/airbytehq/airbyte-internal-issues/issues/10002

As a result, source-mysql now passes CAT tests

Screenshot 2024-10-25 at 9 50 52 PM

## Review guide

This reuses the CAT test facilities our tooling team built when we were adding CAT tests to source-postgres.

## Can this PR be safely reverted and rolled back?

- [X ] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Review by clnoll — CHANGES_REQUESTED** [review]
Nice @theyueli! Great to see more DB sources using CATs.

I focused my review on hook.py and have some requests related to using context managers for resource management.

File: airbyte-integrations/connectors/source-mysql/integration_tests/seed/requirement.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+mysql-connector-python
```
[the remark this answers is not in the document]
Comment (theyueli) [reply airbyte-integrations/connectors/source-mysql/integration_tests/seed/requirement.txt:1]: fixed.

File: airbyte-integrations/connectors/source-mysql/integration_tests/seed/hook.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        for record in records:
+            cursor.execute(insert_query, record)
+
+        conn.commit()
+        print("Records inserted successfully")
+    except Error as error:
+        print(f"Error inserting records: {error}")
+        conn.rollback()
+    finally:
+        cursor.close()
```
[the remark this answers is not in the document]
Comment (theyueli) [reply airbyte-integrations/connectors/source-mysql/integration_tests/seed/hook.py:63]: Thanks for the suggestion, fixed!

File: airbyte-integrations/connectors/source-mysql/integration_tests/seed/hook.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+def create_schema(conn, schema_name: str) -> None:
```
[the remark this answers is not in the document]
Comment (theyueli) [reply airbyte-integrations/connectors/source-mysql/integration_tests/seed/hook.py:65]: fixed.

File: airbyte-integrations/connectors/source-mysql/integration_tests/seed/hook.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+        with open(secret_active_config_file, 'w') as f:
+            json.dump(secret, f)
+
+    with open(secret_config_cdc_file) as base_config:
+        secret = json.load(base_config)
+        secret["database"] = schema_name
+        with open(secret_active_config_cdc_file, 'w') as f:
+            json.dump(secret, f)
+
+def create_table(conn, schema_name: str, table_name: str) -> None:
```
[the remark this answers is not in the document]
Comment (theyueli) [reply airbyte-integrations/connectors/source-mysql/integration_tests/seed/hook.py:104]: fixed.

File: airbyte-integrations/connectors/source-mysql/integration_tests/seed/hook.py [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+    schema_name = load_schema_name_from_catalog()
+    new_records = [
+        ('4', 'four'),
+        ('5', 'five')
+    ]
+    connection = connect_to_db()
+    table_name = 'id_and_name_cat'
+    if connection:
+        insert_records(connection, schema_name, table_name, new_records)
+        connection.close()
```
[the remark this answers is not in the document]
Comment (theyueli) [reply airbyte-integrations/connectors/source-mysql/integration_tests/seed/hook.py:143]: fixed.

File: airbyte-integrations/connectors/source-mysql/integration_tests/seed/hook.py [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+    create_table(connection, schema_name, table_name)
+    insert_records(connection, schema_name, table_name, records)
+    connection.close()
+
+def load_schema_name_from_catalog():
+    with open("./generated_schema.txt", "r") as f:
+        return f.read()
+
+def delete_schemas_with_prefix(conn, date_prefix):
+    try:
```
[the remark this answers is not in the document]
Comment (theyueli) [reply airbyte-integrations/connectors/source-mysql/integration_tests/seed/hook.py:167]: fixed.

File: airbyte-integrations/connectors/source-mysql/integration_tests/seed/hook.py [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+            user=secret["username"],
+            password=secret["password"],
+            host=secret["host"],
+            port=secret["port"]
+        )
+        print("Connected to the database successfully")
+        return conn
+    except Error as error:
+        print(f"Error connecting to the database: {error}")
+        sys.exit(1)
```
[the remark this answers is not in the document]
Comment (theyueli) [reply airbyte-integrations/connectors/source-mysql/integration_tests/seed/hook.py:47]: fixed, thanks!

File: airbyte-integrations/connectors/source-mysql/integration_tests/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,14 @@
+# Use a base image with MySQL client
+FROM python:3.9
+
+WORKDIR /usr/src/app
+
+# Copy the script to the container
+COPY seed/hook.py .
+COPY seed/requirement.txt .
+
+RUN pip install --no-cache-dir -r ./requirement.txt
```
[the remark this answers is not in the document]
Comment (theyueli) [reply airbyte-integrations/connectors/source-mysql/integration_tests/Dockerfile]: just to confirm, you meant the following?

```
COPY seed/requirement.txt .
RUN pip install --no-cache-dir -r ./requirement.txt
COPY seed/hook.py .
```

**Review by clnoll — APPROVED** [review]
Looks great @theyueli!

File: airbyte-integrations/connectors/source-mysql/integration_tests/Dockerfile (hunk 8 above)
[the remark this answers is not in the document]
Comment (theyueli) [reply airbyte-integrations/connectors/source-mysql/integration_tests/Dockerfile]: great, fixed!

**Pull request merged**